udp_bridge/sender.py

```diff
@@ -116,61 +102,81 @@ def validate_params(node:Node):
     if len(node.get_parameter("topics").value) == 0:
         node.get_logger().warn("parameter 'topics' is an empty list")
 
-    if not node.has_parameter('sender_queue_max_size'):
+    if not node.has_parameter("sender_queue_max_size"):
         node.get_logger().fatal("parameter 'sender_queue_max_size' not found")
         result = False
-    if not isinstance(node.get_parameter('sender_queue_max_size').value, int):
+    if not isinstance(node.get_parameter("sender_queue_max_size").value, int):
         node.get_logger().fatal("parameter 'sender_queue_max_size' is not an Integer")
         result = False
 
-    if not node.has_parameter('send_frequency'):
+    if not node.has_parameter("send_frequency"):
         node.get_logger().fatal("parameter 'send_frequency' not found")
         result = False
-    if not isinstance(node.get_parameter('send_frequency').value, float) \
-        and not isinstance(node.get_parameter('send_frequency').value, int):
+    if not isinstance(node.get_parameter("send_frequency").value, float) and not isinstance(
+        node.get_parameter("send_frequency").value, int
+    ):
         node.get_logger().fatal("parameter 'send_frequency' is not an Integer or Float")
         result = False
 
     return result
 
 
-def main():
-    global CIPHER
-    rclpy.init()
-    node = Node('udp_bridge_sender', automatically_declare_parameters_from_overrides=True)
-    thread = Thread(target=rclpy.spin, args=(node,))
-    thread.start()
-    if validate_params(node):
-        encryption_key = None
-        if node.has_parameter("encryption_key"):
-            encryption_key = node.get_parameter("encryption_key").value
-        CIPHER = AESCipher(encryption_key)
-        port = node.get_parameter("port").value
-        freq = node.get_parameter("send_frequency").value
-        targets = node.get_parameter('target_ips').value
-        max_queue_size = node.get_parameter('sender_queue_max_size').value
-        topics = node.get_parameter("topics").value
+class UdpBridgeSender:
+    def __init__(self, node: Node):
+        self.node = node
+        self.freq: float = node.get_parameter("send_frequency").value
+
+        params_blackboard = get_parameters_from_other_node(node, "parameter_blackboard", ["monitoring_host_ip"])
+        self.target: str = params_blackboard["monitoring_host_ip"]
+        self.port: int = node.get_parameter("port").value
+        self.sock = self.setup_udp_broadcast_socket()
 
+        topics: list[str] = node.get_parameter("topics").value
+        max_queue_size: int = node.get_parameter("sender_queue_max_size").value
+        message_handler = self.setup_message_handler()
+        self.subscribers: list[AutoSubscriber] = list(
+            map(lambda topic: AutoSubscriber(topic, max_queue_size, message_handler, node), topics)
+        )
+
+    def setup_udp_broadcast_socket(self) -> socket.socket:
```

Comment:
Why do we want to broadcast? Don't we wanna send only to the target ip? Or is this just necessary socket-lingo?